### ISSUE TITLE
[backend] "Updating" a recurring run, whilst a recurring run is in progress causes it to terminate

### ISSUE THREAD

**Issue by alexlatchford** [kubeflow/pipelines#7421]
### Environment

*  How did you deploy Kubeflow Pipelines (KFP)?

Atop AWS using the `kubeflow/manifests` official distribution with some overlays specific to Zillow.

*  KFP version: 

We use KF v1.3.1 currently (plan to update to v1.5 next month), looks like that includes KFP v1.5.1 ([link](https://github.com/kubeflow/manifests/blob/v1.3.1/apps/pipeline/upstream/base/pipeline/kustomization.yaml#L43)).

*  KFP SDK version: 

We run a forked version ([see here](https://github.com/zillow/pipelines)), I think we last pulled in at ~v1.8.

### Steps to reproduce

The problem is that via CICD we want our scientists to be able to define a `recurring-run` and have it updated on subsequent CICD executions if they change their settings. Given the limitations of the "Job" API in KFP we delete and re-add to "update" a recurring run, (see the [API reference](https://www.kubeflow.org/docs/components/pipelines/reference/api/kubeflow-pipeline-api-spec/#tag-JobService) for more info).

What happens in this case if the first recurring run has triggered a workflow then the scheduledworkflow controller deletes the `ScheduledWorkflow` and then via Kubernetes garbage collection it cascades and deletes any in-progress `Workflow` resources. Then when the "new" updated recurring run is created it needs to wait for the new trigger point to schedule the next run, it won't try to resurrect the now terminated run (and it definitely shouldn't thinking about it).

### Expected result

Basically some way of keeping the run that was in progress before the "update" occurred.

Potentially when deleting a recurring run if there was a flag available in the API/SDK/CLI to say to [orphan on deletion](https://kubernetes.io/docs/tasks/administer-cluster/use-cascading-deletion/#set-orphan-deletion-policy) that'd solve this issue and you'd not need a complicated update API.

### Materials and Reference

Apologies, maybe this is more of a "feature" request thinking about it now. Apologies for the mischaracterization!

---

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by tleewongjaro-agoda** [comment]
@alexlatchford  Hello, we had the same problem. So we change behavior here: https://github.com/kubeflow/pipelines/blob/master/backend/src/apiserver/resource/resource_manager.go#L1063

By setting DeleteOption to include DeletePropagationOrphan

```
	propagationPolicy := v1.DeletePropagationOrphan
	err = r.getScheduledWorkflowClient(job.Namespace).Delete(ctx, job.Name, &v1.DeleteOptions{
		PropagationPolicy: &propagationPolicy,
	})
```

But it'd be better if this can be provided as a flag on rest api as well.
